editor/editor_themes.cpp

```diff
@@ -275,36 +303,101 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
+	style_popup->set_border_color_all(contrast_color_1);
+	style_popup->set_border_width_all(MAX(EDSCALE, border_width));
+	const Color shadow_color = Color(0, 0, 0, dark_theme ? 0.3 : 0.1);
+	style_popup->set_shadow_color(shadow_color);
+	style_popup->set_shadow_size(4 * EDSCALE);
+
+	Ref<StyleBoxEmpty> style_empty = make_empty_stylebox(default_margin_size, default_margin_size, default_margin_size, default_margin_size);
+
+	// Tabs
+	Ref<StyleBoxFlat> style_tab_selected = style_default->duplicate();
+	style_tab_selected->set_default_margin(MARGIN_LEFT, 10 * EDSCALE);
+	style_tab_selected->set_default_margin(MARGIN_RIGHT, 10 * EDSCALE);
+	style_tab_selected->set_bg_color(tab_color);
+
+	Ref<StyleBoxEmpty> style_tab_unselected = style_empty->duplicate();
+	style_tab_unselected->set_default_margin(MARGIN_LEFT, 10 * EDSCALE);
+	style_tab_unselected->set_default_margin(MARGIN_RIGHT, 10 * EDSCALE);
+
 	// Editor background
-	theme->set_stylebox("Background", "EditorStyles", make_flat_stylebox(dark_color_2, 4, 4, 4, 4));
+	theme->set_stylebox("Background", "EditorStyles", make_flat_stylebox(background_color, default_margin_size, default_margin_size, default_margin_size, default_margin_size));
 
 	// Focus
-	Ref<StyleBoxFlat> focus_sbt = make_flat_stylebox(contrast_color_1, 4, 4, 4, 4);
-	focus_sbt->set_draw_center(false);
-	focus_sbt->set_border_width_all(1 * EDSCALE);
-	focus_sbt = change_border_color(focus_sbt, contrast_color_2);
-	theme->set_stylebox("Focus", "EditorStyles", focus_sbt);
+	Ref<StyleBoxFlat> style_focus = style_default->duplicate();
+	style_focus->set_draw_center(false);
+	style_focus->set_border_color_all(contrast_color_2);
+	theme->set_stylebox("Focus", "EditorStyles", style_focus);
 
 	// Menu
-	Ref<StyleBoxEmpty> style_menu = make_empty_stylebox(4, 4, 4, 4);
+	Ref<StyleBoxEmpty> style_menu = style_empty;
 	theme->set_stylebox("panel", "PanelContainer", style_menu);
 	theme->set_stylebox("MenuPanel", "EditorStyles", style_menu);
 
 	// Play button group
-	theme->set_stylebox("PlayButtonPanel", "EditorStyles", make_empty_stylebox(8, 4, 8, 4)); //make_stylebox(theme->get_icon("GuiPlayButtonGroup", "EditorIcons"), 16, 16, 16, 16, 8, 4, 8, 4));
+	theme->set_stylebox("PlayButtonPanel", "EditorStyles", style_empty); //make_stylebox(theme->get_icon("GuiPlayButtonGroup", "EditorIcons"), 16, 16, 16, 16, 8, 4, 8, 4));
 
 	//MenuButton
-	Ref<StyleBoxFlat> style_menu_hover_border = make_flat_stylebox(highlight_color, 4, 4, 4, 4);
-	Ref<StyleBoxFlat> style_menu_hover_bg = make_flat_stylebox(dark_color_2, 4, 4, 4, 4);
-
+	Ref<StyleBoxFlat> style_menu_hover_border = style_default->duplicate();
 	style_menu_hover_border->set_draw_center(false);
-	style_menu_hover_border->set_border_width(MARGIN_BOTTOM, 2 * EDSCALE);
-	style_menu_hover_border->set_border_color_all(highlight_color);
+	style_menu_hover_border->set_border_width_all(0);
+	style_menu_hover_border->set_border_width(MARGIN_BOTTOM, border_width * EDSCALE);
```

Comment:
I would give that ` MAX(border_width * EDSCALE, EDSCALE)
so that there is always a border even if turned of?

Reply:
@toger5 yeah that could work too